src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -859,8 +858,10 @@ private void InitializePackagesFolderInfo(string packagesFolderPath, bool exclud
                 {
                     var installedPackageReference = projectInstalledPackageReferences
                         .FirstOrDefault(pr => StringComparer.OrdinalIgnoreCase.Equals(pr.PackageIdentity.Id, packageId));
+                    // Skip autoreferenced update when we have only a package ID.
+                    var autoReferenced = IsPackageReferenceAutoReferenced(installedPackageReference);
 
-                    if (installedPackageReference != null)
+                    if (installedPackageReference != null && !autoReferenced)
```

Comment:
Is the issue just about autoReferenced packages or can this be applied to regular package updates which are blocked by project?

Reply:
"autoreferenced" packages are only ones that we support blocking in PR. 

This fix only changes auto referenced packages.